conans/client/downloaders/download_cache.py

```diff
@@ -73,22 +73,33 @@ def should_upload_sources(package):
                 if ref.get("upload") or any(should_upload_sources(p) for p in packages):
                     all_refs.add(str(k))
 
-        files_to_upload = []
+        path_backups_contents = []
 
+        dirty_ext = ".dirty"
         for path in os.listdir(path_backups):
+            if remove_if_dirty(os.path.join(path_backups, path)):
+                continue
+            if path.endswith(dirty_ext):
+                # TODO: Clear the dirty file marker if it does not have a matching downloaded file
```

Comment:
When we rethink how the `.dirty` files are handled, we won't need to check here nor do anything special